client/my-sites/plugins/search-box-header/index.jsx

```diff
@@ -8,6 +9,7 @@ import { recordGoogleEvent, recordTracksEvent } from 'calypso/state/analytics/ac
 import './style.scss';
 
 function pageToSearch( s ) {
+	page.show( page.current );
```

Comment:
Could you add a comment here explaining why this is needed when setQueryArgs also updates location.href? I would suggest something but I don't think I understand this well enough to explain it properly.